src/domain/gift-cards/manage/index.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  useAdminProduct,
+  useAdminProductTypes,
+  useAdminStore,
+} from "medusa-react"
+import React, { useEffect, useState } from "react"
+import { useForm } from "react-hook-form"
+import PlusIcon from "../../../components/fundamentals/icons/plus-icon"
+import TrashIcon from "../../../components/fundamentals/icons/trash-icon"
+import UnpublishIcon from "../../../components/fundamentals/icons/unpublish-icon"
+import StatusDot from "../../../components/fundamentals/status-indicator"
+import BreadCrumb from "../../../components/molecules/breadcrumb"
+import Input from "../../../components/molecules/input"
+import Select from "../../../components/molecules/select"
+import TagInput from "../../../components/molecules/tag-input"
+import BodyCard from "../../../components/organisms/body-card"
+import DetailsCollapsible from "../../../components/organisms/details-collapsible"
+import DenominationTable from "./denomination-table"
+
+type ManageGiftCardProps = {
+  path: string
+  id: string
+  updateGiftCard: (data) => void
+  deleteGiftCard: () => void
+}
+
+const ManageGiftCard: React.FC<ManageGiftCardProps> = ({
+  id,
+  updateGiftCard,
+  deleteGiftCard,
+}: ManageGiftCardProps) => {
+  const { store } = useAdminStore()
+  const { product: giftCard, isSuccess } = useAdminProduct(id)
+  const { types } = useAdminProductTypes()
+
+  const { register, handleSubmit, reset } = useForm()
+
+  const [type, setType] = useState<{ label: string; value: string } | null>(
+    giftCard?.type
+      ? { value: giftCard?.type.value, label: giftCard?.type.value }
+      : null
+  )
+  const [tags, setTags] = useState<string[]>(
+    giftCard?.tags?.map((t) => t.value) || []
+  )
+
+  register("title")
+  register("subtitle")
+  register("description")
+
+  const submit = (data) => {
+    const update = { ...data }
+
+    if (typeof type === `undefined`) {
```

Comment:
type is null if you switch between publish and unpublish in the body card which causes errors